backend/tests/test_entry_compliance.py

```
@@ -0,0 +1,556 @@
+            trade_id="trade-1",
+            evaluated_at="2026-01-16T10:00:00+00:00",
+            dte_at_entry=31,
+            delta_at_entry=0.25,
+            monthly_return_pct=2.5,
+            earnings_buffer_days=None,
+            compliant=1,
+            failed_rules=json.dumps([]),
+            entry_rules_snapshot=json.dumps({}),
+        )
+    )
+    with pytest.raises(Exception):  # IntegrityError from SQLite PK violation
```

Ruff (0.15.13)

[warning] Do not assert blind exception: `Exception`

(B017)